ansible/extras.yml

```diff
@@ -1,3 +1,25 @@
+- hosts: k3s_server
+  become: yes
+  tags: k3s
+  tasks:
+    - name: Start k3s server
+      when: "'builder' not in group_names"
+      ansible.builtin.include_role:
+        name: k3s
+        tasks_from: server-runtime.yml
+
+- hosts: k3s_agent
+  become: yes
+  tags: k3s
+  tasks:
+    - name: Start k3s agents
+      when: "'builder' not in group_names"
+      vars: # set outside of role to allow compute init to define own value
+        k3s_bootstrap_token: "{{ hostvars[groups['k3s_server'] | first]._k3s_token_output.stdout | default('') }}"
```

Comment:
No. Using task vars is bad, b/c a) its yet another precedence level to think about and b) you can't override them in the appliance.

Can we not explicitly set_fact for the k3s_bootstrap_token on these hosts?

Reply:
I think what I want to have happen is:

1. The `server-runtime.yml` play gets an extra task:
```
- set_fact:
     k3s_bootstrap_token: ._k3s_token_output.stdout
```

2. That's set as a default via the environment:
```environments/common/group_vars/all/k3s.yml:
k3s_bootstrap_token: "{{ hostvars[groups['k3s_server'] | first].k3s_bootstrap_token | default('') }}"
```

or maybe, default None is safer than accidently providing an empty token somehow??

I guess I'm not clear if tokens can *ever* apply to the k3s-server too?

Reply:
Really, I'd like to move this all back to bootstrap.yml too (late on), i.e. after that we have a "working cluster". To me extras.yml should probably be for optional extras. And monitoring.yml is going to require k3s. So the other choice is to move it there, but I think that's not where we want it long-term.